tests/subsys/logging/dictionary/pytest/test_logging_dictionary.py

```diff
@@ -33,13 +33,21 @@ def process_logs(dut: DeviceAdapter, build_dir):
     logger.info(f'Log parser script: {parser_script}')
 
     # And also the dictionary JSON file is there...
-    dictionary_json = os.path.join(build_dir, "zephyr", "log_dictionary.json")
+    is_sysbuild_dir = os.path.join(build_dir, "zephyr", "dictionary")
+
+    if os.path.isdir(is_sysbuild_dir):
+        # With sysbuild, the test app is built under another layer of directory
+        # where the JSON file is.
+        dictionary_json = os.path.join(build_dir, "zephyr", "dictionary", "log_dictionary.json")
+    else:
+        dictionary_json = os.path.join(build_dir, "zephyr", "log_dictionary.json")
+
```

Comment:
look into https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/pytest-twister-harness/src/twister_harness/twister_harness_config.py#L42

you can use as `dut.device_config.app_build_dir` in `test_logging_dictionary` to have both sysbuild and non-sysbuild scenarios covered

Reply:
That's good to know. Thanks.